STUDIO/org.openl.rules.webstudio/src/org/openl/rules/rest/UsersService.java

```diff
@@ -0,0 +1,287 @@
+            if (authentication.getPrincipal() instanceof SimpleUser) {
+                return (SimpleUser) authentication.getPrincipal();
+            } else {
+                return (SimpleUser) authentication.getDetails();
+            }
+        }).ifPresent(simpleUser -> {
+            simpleUser.setFirstName(firstname);
+            simpleUser.setLastName(lastname);
+            simpleUser.setEmail(email);
+            simpleUser.setDisplayName(displayName);
+            if (StringUtils.isNotEmpty(newPassword)) {
+                simpleUser.setPassword(passwordEncoder.encode(newPassword));
+            }
+        });
+    }
+
+    @GET
+    @Path("/profile")
+    public UserProfileModel getUserProfile() {
+        String username = currentUserInfo.getUserName();
+        User user = Optional.ofNullable(currentUserInfo.getAuthentication()).map(authentication -> {
+            if (authentication.getPrincipal() instanceof User) {
+                return (User) authentication.getPrincipal();
+            } else if (authentication.getDetails() instanceof User) {
+                return (User) authentication.getDetails();
+            } else {
+                return userManagementService.getApplicationUser(username);
+            }
+        })
+            .orElse(new SimpleUser(null,
+                null,
+                username,
+                null,
+                Collections.emptyList(),
+                null,
+                null,
+                new UserExternalFlags()));
+
+        return new UserProfileModel().setFirstName(user.getFirstName())
+            .setLastName(user.getLastName())
+            .setEmail(user.getEmail())
+            .setShowHeader(IXlsTableNames.VIEW_DEVELOPER
+                .equals(userSettingsManager.getStringProperty(user.getUsername(), TABLE_VIEW)))
+            .setShowFormulas(userSettingsManager.getBooleanProperty(user.getUsername(), TABLE_FORMULAS_SHOW))
+            .setTestsPerPage(userSettingsManager.getIntegerProperty(user.getUsername(), TEST_TESTS_PERPAGE))
+            .setTestsFailuresOnly(userSettingsManager.getBooleanProperty(user.getUsername(), TEST_FAILURES_ONLY))
+            .setTestsFailuresPerTest(userSettingsManager.getIntegerProperty(user.getUsername(), TEST_FAILURES_PERTEST))
+            .setShowComplexResult(userSettingsManager.getBooleanProperty(user.getUsername(), TEST_RESULT_COMPLEX_SHOW))
+            .setShowRealNumbers(userSettingsManager.getBooleanProperty(user.getUsername(), TRACE_REALNUMBERS_SHOW))
+            .setDisplayName(user.getDisplayName())
+            .setUsername(user.getUsername())
+            .setInternalUser(StringUtils.isNotBlank(user.getPassword()))
+            .setExternalFlags(user.getExternalFlags());
+    }
+
+    @DELETE
+    @Path("/{username}")
+    public void deleteUser(@PathParam("username") String username) {
+        userManagementService.deleteUser(username);
```

Comment:
throw NotFoundException if user with the same doesn't exist

Reply:
Fixed.